File: Year-2023/LeetCode-May-Challenge/MaximumTwinSumofaLinkedList.js
function ListNode(val, next) {
    this.val = (val === undefined ? 0 : val)
    this.next = (next === undefined ? null : next)
}

var pairSum = function (head) {
    let list = new Array();
    let temp = head;
    while (temp) {
        list.push(temp.val);
        temp = temp.next;
    }
    let res = 0, n = list.length;
    for (let i = 0; i < n / 2; ++i) res = Math.max(res, list[i] + list[n - i - 1]);
    return res;
};

File: Year-2023/LeetCode-May-Challenge/MaximumTwinSumofaLinkedList.java
import java.util.*;

class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}
class Solution {
    public int pairSum(ListNode head) {
        ArrayList<Integer> list = new ArrayList<>();
        ListNode temp = head;
        while(temp != null){
            list.add(temp.val);
            temp = temp.next;
        }
        int res = 0, n = list.size();
        for(int i = 0; i < n / 2; ++i){
            res = Math.max(res, list.get(i) + list.get(n - i - 1));
        }
        return res;
    }
}